drakrun/drakrun/analyzer.py

```diff
@@ -0,0 +1,515 @@
+        formatter_class=argparse.ArgumentDefaultsHelpFormatter,
+    )
+    parser.add_argument("sample_path", help="Path to the sample")
+    parser.add_argument(
+        "--vm-id",
+        help="Virtual machine ID to use for analysis",
+        type=int,
+        required=True,
+    )
+    parser.add_argument(
+        "--output-dir",
+        help="Path where analysis results will be written",
+        required=True,
+    )
+    parser.add_argument(
+        "--plugins",
+        help="DRAKVUF plugins to enable during analysis",
+        default=drakconfig.drakvuf_plugins.get_plugin_list(),
+        action=PluginsArgAction,
+    )
+    parser.add_argument(
+        "--timeout",
+        default=drakconfig.drakrun.analysis_timeout,
+        type=int,
+        help="Timeout in seconds for analysis",
+    )
+    parser.add_argument(
+        "--hooks-path", help="Path to the custom hook list", required=False
+    )
+    parser.add_argument(
+        "--start-command", help="Alternative command to start analysis", required=False
+    )
+    parser.add_argument(
+        "--extension",
+        help="Alternative extension indicating sample format",
+        required=False,
+    )
+    parser.add_argument(
+        "--sample-filename", help="Target file name for sample", required=False
+    )
+    parser.add_argument(
+        "--dns-server",
+        help="DNS server to use for analysis",
+        default=drakconfig.drakrun.dns_server,
+        required=False,
+    )
+    parser.add_argument(
+        "--out-interface",
+        help="Interface to be used by VM for Internet communication",
+        default=drakconfig.drakrun.out_interface,
+        required=False,
+    )
+    parser.add_argument(
+        "--net-enable",
+        help="If enabled, VM will be able to connect to the Internet via out-interface",
+        default=drakconfig.drakrun.net_enable,
+        action=argparse.BooleanOptionalAction,
+    )
+    parser.add_argument(
+        "--anti-hammering-threshold",
+        help="Threshold for API hammering detection (detection disabled if 0)",
+        default=drakconfig.drakrun.anti_hammering_threshold,
+        type=int,
+    )
+    parser.add_argument(
+        "--syscall-filter",
+        help="Syscall filter for syscalls plugin",
+        default=drakconfig.drakrun.syscall_filter,
+        type=str,
+    )
+    parser.add_argument(
+        "--raw-memory-dump",
+        help="Make full memory dump of VM after analysis",
+        default=drakconfig.drakrun.raw_memory_dump,
+        action=argparse.BooleanOptionalAction,
+    )
+
+    args = parser.parse_args()
+
+    sample_path = pathlib.Path(args.sample_path)
+    if not sample_path.exists():
+        raise RuntimeError(f"Provided file '{str(args.sample_path)}' does not exist")
+
+    output_dir = pathlib.Path(args.output_dir)
+    if output_dir.exists():
+        confirmation = ["Y", "y", ""]
+        if (
+            input("Output directory already exists. Overwrite? (Y/n)")
```

Comment:
I think i would actually loop here until user presses either `y` or `n` :thinking:. My reasoning is - the consequence of accidentally removing a live folder are pretty severe, and typing `y[enter]` shouldn't be a problem for user.

And also in both cases it would be useful to have -f flag, in case someone runs this tool from non-interactive command line.

Reply:
Done